lib/galaxy/tools/__init__.py

```diff
@@ -2772,10 +2773,8 @@ def json_fix( val ):
         return [ json_fix( v ) for v in val ]
     elif isinstance( val, dict ):
         return dict( [ ( json_fix( k ), json_fix( v ) ) for ( k, v ) in val.iteritems() ] )
-    elif isinstance( val, unicode ):
-        return val.encode( "utf8" )
     else:
-        return val
+        return unicodify(val, "utf8" )
```

Comment:
right, I was thinking it might be advantageous to use unicode here and [here] (https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/json.py#L24)
It did prevent issue #1702 for me (in a different context, of course), ...